src/lib.rs

```diff
@@ -24,138 +20,66 @@ pub mod termination;
 pub mod util;
 pub mod view;
 
-use crate::draft::*;
-use crate::event::*;
-use crate::lockup::*;
-use crate::schedule::*;
-use crate::termination::*;
-use crate::util::*;
-
-near_sdk::setup_alloc!();
+use crate::{events::*, lockup::*, schedule::*, util::*};
 
-pub type TimestampSec = u32;
 pub type TokenAccountId = AccountId;
 
 pub const PACKAGE_NAME: &str = env!("CARGO_PKG_NAME");
 pub const VERSION: &str = env!("CARGO_PKG_VERSION");
 
-const GAS_FOR_FT_TRANSFER: Gas = 15_000_000_000_000;
-const GAS_FOR_AFTER_FT_TRANSFER: Gas = 20_000_000_000_000;
-const GAS_EXT_CALL_COST: Gas = 10_000_000_000_000;
-const GAS_MIN_FOR_CONVERT: Gas = 15_000_000_000_000;
-
-const ONE_YOCTO: Balance = 1;
-const NO_DEPOSIT: Balance = 0;
-
-uint::construct_uint! {
-    pub struct U256(4);
-}
-
-#[ext_contract(ext_self)]
-pub trait SelfCallbacks {
-    fn after_ft_transfer(
-        &mut self,
-        account_id: AccountId,
-        lockup_claims: Vec<LockupClaim>,
-    ) -> WrappedBalance;
+const GAS_FOR_FT_TRANSFER: Gas = Gas::from_gas(15_000_000_000_000);
+const GAS_FOR_AFTER_FT_TRANSFER: Gas = Gas::from_gas(20_000_000_000_000);
 
-    fn after_lockup_termination(
-        &mut self,
-        account_id: AccountId,
-        amount: WrappedBalance,
-    ) -> WrappedBalance;
-
-    fn convert_drafts(&mut self, draft_ids: Vec<DraftIndex>) -> Vec<LockupIndex>;
-}
+const ONE_YOCTO: NearToken = NearToken::from_yoctonear(1);
 
-#[near_bindgen]
-#[derive(BorshDeserialize, BorshSerialize, PanicOnDefault)]
+#[near(contract_state, serializers = [borsh])]
+#[derive(PanicOnDefault)]
 pub struct Contract {
-    pub token_account_id: TokenAccountId,
+    pub token_account_id: AccountId,
 
     pub lockups: Vector<Lockup>,
 
-    pub account_lockups: LookupMap<AccountId, HashSet<LockupIndex>>,
+    pub account_lockups: LookupMap<AccountId, UnorderedSet<LockupIndex>>,
 
     /// account ids that can perform all actions:
-    /// - manage deposit_whitelist
-    /// - manage drafts, draft_groups
-    /// - create lockups, terminate lockups, fund draft_groups
-    pub deposit_whitelist: UnorderedSet<AccountId>,
-
-    /// account ids that can perform all actions on drafts:
-    /// - manage drafts, draft_groups
-    pub draft_operators_whitelist: UnorderedSet<AccountId>,
-
-    pub next_draft_id: DraftIndex,
-    pub drafts: LookupMap<DraftIndex, Draft>,
-    pub next_draft_group_id: DraftGroupIndex,
-    pub draft_groups: UnorderedMap<DraftGroupIndex, DraftGroup>,
+    /// - manage deposit_allowlist
+    /// - create lockups, terminate lockups
+    pub deposit_allowlist: UnorderedSet<AccountId>,
```

Comment:
I want to highlight for anyone reviewing this code in the future that there is a breaking change in the STATE layout, so this version of the contract cannot be deployed over the old version of the contract without a migration